staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go

```diff
@@ -23,49 +23,33 @@ import (
 	"k8s.io/apiserver/pkg/authentication/user"
 )
 
+// Authenticator is the umbrella interface for all authenticators.
+type Authenticator interface {
+	AuthenticatorID() string
```

Comment:
a peer ID method won't work for the kind of union authentication we do. I think you need to add an authenticator field to the Response struct instead of broadening the Authenticator interfaces

Reply:
Thanks @liggitt 
I will take another stab at this on another PR - will try the path of extending the Response. I think we will also need to return a custom error (an error that contains the ID of a failed authenticator), this is so that we can instrument failed calls as well.